pyresttest/validators.py

```diff
@@ -289,6 +290,24 @@ def parse(cls, config, extractor_base=None):
         return cls.configure_base(config, base)
 
 
+class CookieExtractor(AbstractExtractor):
+    """ Extractor that pulls out a named cookie """
+    extractor_type = 'cookie'
+    is_header_extractor = True
+
+    def extract_internal(self, query = None, args = None, body = None, headers = None):
+        extracted = [y[1] for y in filter(lambda x: x[0] == 'set-cookie', headers) if y[1].startswith(query + '=')]
+        if len(extracted) == 0:
+            raise ValueError("Cookie named {0} not known".format(query))
+        else:
+            return Cookie.SimpleCookie(extracted[-1])[query].value
```

Comment:
> it return the last index of cookies. The behaviors may be strange.

@alswl What do you mean by this?  The -1 array index is fine, and we *should* be using the last cookie specified, I'm just wondering about the Cookie.SimpleCookie there and what its failure modes are. 

I think you also meant this, which doesn't end up any different than the original in truth (it is slightly cleaner):

```python
if len(extracted) == 0:
    raise ValueError('xxx')
return Cookie.SimpleCookie(extracted[-1])[query].value 
```